proposal.rst

```diff
@@ -272,34 +287,138 @@ It would still produce better estimates as we reduced the number of documents we
 
- 	* Build some tests on debug mode to see how functions are invoked.
+	This is treated like MULTI_AND
 
- 	* Search more for ways to tackle this task.
+	let's say L FILTER Right
 
-* Two Weeks: Working on OP_AND, OP_OR, and OP_AND_NOT (Code, testing, and doumentation)
+	then we just return probability of L times the probability of R times the total range, which would be the union of the two ranges.
 
+NEAR, and PHRASE :
 
-* One and half Week: Working on OP_XOR,and OP_AND_MAYBE (Code, testing, and doumentation.)
+	No work needed on this.
 
+VALUE_RANGE :
+
+	We can make use of the fact that we have the first and last docids, to reduce the total range to be the overlapped range of the
+	begin and end ranges, instead of the slot_freq reange.
+
+VALUE_GT, VALUE_LT, VALUE_LE, VALUE_GE :
+
+	implementing get_used_docid_range().
+
+	We can just return the range of the first, and last docids.
+
+ELITE_SET :
+
+	implementing get_used_docid_range().
+
+	Get the elite set of N terms, then treat it like MULTI_OR
+
+SYNONYM :
+
+	implementing get_used_docid_range().
+
+	treated like OP_OR
+MAX :
+
+	implementing get_used_docid_range();
+
+	treated like OP_OR
+
+WILDCARD, EDIT_DISTANCE :
+
+	implementing get_used_docid_range().
+
+	This would return the matches terms along with them ranges,then we can calucate as OP_OR;
+
+INVALID, LEAF_TERM, LEAF_POSTING_SOURCE, LEAF_MATCH_ALL,LEAF_MATCH_NOTHING, OP_SCALE_WEIGHT :
+
+	Unrelated.
+
+|
+
+My initial plan is as follow:
+
+	**Every week would be 3 days of coding, 2 days of testing and documenting.**
+
+	**BY the end of every week, new edits should be ready to be merged.**
+|
+
+First three weeks of bonding :
+
+	week one and two : get familiar with the Xapian::matcher code base.
+
+	week three : setup debug environment, and how to test project parts, read c++ concepts on developers guide.
+week 1 : Work on OP_AND_NOT
+
+week 2 : Work on OP_XOR
+
+week 3 : Work on OP_AND_MAYBE
+
+week 4 : Work on OP_FILTER
 
 10 June : Final exams (this is an estimate,official schedule isn't announced yet.)
 
+First Evaluation
+
+week 5 : Work on OP_ELITE_SET
+
+week 6 : Work on OP_SYNONYM
 
-* Two Weeks: Working on OP_FILTER, OP_NEAR, and OP_PHRASE (Code, testing, and doumentation.)
+week 7 : Work on OP_MAX
 
-* Two Weeks: Working on OP_VALUE_RANGE, OP_VALUE_LE, OP_VALUE_LT, OP_VALUE_GE, and OP_VAUE_GT (Code, testing, and doumentation).
+week 8 : Work on OP_WILDCARD, and OP_EDIT_DISTANCE
 
-* Two Weeks: Working on OP_SCALE_WEIGHT,and OP_ELITE_LIST (Code, testing, and doumentation.)
+Second Evaluation
 
-* One Weeks: Working on OP_SYNONYM,and OP_MAX (Code, testing, and doumentation.)
+week 9 : Work on OP_VALUE_RANGE
 
-* One Weeks: Working on OP_WILDCARD,and OP_INVALID (Code, testing, and doumentation.)
+week 10 : Work on OP_VALUE_GT, OP_VALUE_GE, OP_VALUE_LT, and OP_VALUE_LE
 
-This could be done in a shorter amount of time. If so, i can start work on other projects.
+week 11 , 12 : Review, see if anything need to change.
```

Comment:
That's rather a nebulous task for two weeks.

It's good to allow for things potentially overrunning, but a better way to handle this is to identify some "stretch goals" which you'd do if time allows, but would be skipped if things overrun.

The project idea even suggests something else that could be looked at:

> if we ran some experiments using real datasets and associated queries we could look at how the estimates compare to the true answers, and use the results to tune the calculations of the estimate to try to reduce the skewing effect we get from assuming independence.

But you're welcome to think up useful things for yourself - for example, your idea of storing some sort of co-occurrence data from a standard dataset (a challenge there seems to be storing it compactly enough for the cost not to be too high for the benefit, but perhaps we can just store data for cases which differ much from independence or something like that).